core/templates/pages/exploration-editor-page/improvements-tab/services/improvement-learner-answer-details-modal.controller.ts

```diff
@@ -0,0 +1,93 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Controller for ImprovementLearnerAnswerDetailsModal.
+ */
+
+angular.module('oppia').controller(
+  'ImprovementLearnerAnswerDetailsModalController', [
+    '$scope', '$uibModalInstance', 'DateTimeFormatService',
+    'ExplorationHtmlFormatterService', 'LearnerAnswerDetailsDataService',
+    'UserExplorationPermissionsService', 'learnerAnswerDetails',
+    function(
+        $scope, $uibModalInstance, DateTimeFormatService,
+        ExplorationHtmlFormatterService, LearnerAnswerDetailsDataService,
+        UserExplorationPermissionsService, learnerAnswerDetails) {
+      UserExplorationPermissionsService.getPermissionsAsync()
+        .then(permissions => {
```

Comment:
Please pass this in like you do `learnerAnswerDetails`.

Reply:
Done.